sol/main.c

```diff
@@ -376,16 +376,19 @@ iface_speed_bytes(struct gatekeeper_if *iface, uint64_t *link_speed_bytes)
 {
 	uint64_t link_speed_mbits = 0;
 	uint8_t i;
+	int ret;
 
 	RTE_VERIFY(link_speed_bytes != NULL);
 
 	for (i = 0; i < iface->num_ports; i++) {
 		struct rte_eth_link link;
-		rte_eth_link_get(iface->ports[i], &link);
+		ret = rte_eth_link_get(iface->ports[i], &link);
+		if (ret < 0)
+			goto err;
```

Comment:
Log the error of `rte_eth_link_get()` before `goto err;`. Otherwise, a network administrator can't find out what's happening here.